open-metadata-implementation/access-services/analytics-modeling/analytics-modeling-server/src/main/java/org/odpi/openmetadata/accessservices/analyticsmodeling/server/AnalyticsModelingRestServices.java

```diff
@@ -225,6 +226,39 @@ public AnalyticsModelingOMASAPIResponse createArtifact(String serverName, String
 		return ret;
 	}
 
+	/**
+	 * Update analytics artifact defined as json input.
+	 * @param serverName where to create artifact.
+	 * @param userId requested the operation.
+	 * @param serverCapability source where artifact persist.
+	 * @param artifact definition.
+	 * @return response with artifact or error description.
+	 */
+	public AnalyticsModelingOMASAPIResponse updateArtifact(String serverName, String userId, String serverCapability, AnalyticsAsset artifact) {
+
+		String methodName = "updateArtifact";
+		AnalyticsModelingOMASAPIResponse ret;
+		RESTCallToken token = restCallLogger.logRESTCall(serverName, userId, methodName);
+
+		try {
+
+			validateUrlParameters(serverName, userId, null, null, null, null, methodName);
+
+			AssetsResponse response = new AssetsResponse();
+			ResponseContainerAssets assets = getHandler().getAnalyticsArtifactHandler(serverName, userId, methodName)
+					.updateAssets(userId, serverCapability, artifact);
+			response.setAssetList(assets);
+			ret = response;
+		} catch (AnalyticsModelingCheckedException e) {
+			ret = handleErrorResponse(e, methodName);
+		} catch (InvalidParameterException e) {
+			ret = handleInvalidParameterResponse(e, methodName);
+		}
```

Comment:
Need to catch Exception, log the stack trace - using auditLog.logException().  There are helper classes in the ffdc-services module.  (Basically the server should never export a stack trace - this is why we load exceptions into the response objects - the exception is recreated on the client side)
Also need to catch UserNotAuthorizedException ?

Reply:
UserNotAuthorizedException is not thrown. It caught and handled as AnalyticsModelingCheckedException .
Another catch is added for Exception.

Reply:
How does the caller detect that thhey have a security access problem as opposed to a server error etc?